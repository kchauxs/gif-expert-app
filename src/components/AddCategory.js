import React, { useState } from 'react'
import PropTypes from 'prop-types'


export const AddCategory = ({ setCategoties }) => {

    const [inputValue, setInputValue] = useState("")

    const handleInputChange = (e) => setInputValue(e.target.value)

    const handleSumit = (e) => {
        e.preventDefault()

        if (inputValue.trim().length > 2 ) {
            setCategoties(categories => [ inputValue,...categories])
        }
        setInputValue("")
    }

    return (
        <form onSubmit={handleSumit}>
            <input type="text"
                value={inputValue}
                onChange={handleInputChange}
                placeholder="Add Category" />
        </form>
    )
}

AddCategory.protoType = {
    setCategoties:PropTypes.func.isRequired         
}
