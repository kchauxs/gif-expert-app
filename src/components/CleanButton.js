import React, { Fragment } from 'react'
import PropTypes from 'prop-types'


export const CleanButton = ({ setCategoties }) => {

    const handleClear = () => setCategoties([])

    return (
        <Fragment>
            <button onClick={handleClear} className="btn-clean">Clear </button>
        </Fragment>
    )
}

CleanButton.protoType = {
    setCategoties: PropTypes.func.isRequired
}
