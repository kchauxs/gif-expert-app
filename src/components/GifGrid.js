import React, { Fragment } from 'react'
import { useFetchGif } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import logo from '../logo.svg';
// import { getGifs } from '../helpers/getGisf'

export const GifGrid = ({ category }) => {
    
    // const [images, setImages] = useState([])

    // useEffect(() => {
    //     getGifs(category)
    //         .then(setImages)
    // }, [category])


    const { data:images, loading } = useFetchGif(category)


    return (
        <Fragment>
            <h3 className="animate__bounceIn">{category}</h3>

            {/* {loading && <p>Cargando...</p>} */}
            {loading &&  <img src={logo} className="App-logo" alt="logo" />}

            <div className="card-grid">

                {
                    images.map(img => <GifGridItem key={img.id} {...img} />)
                }
            </div>
        </Fragment>
    )
}
