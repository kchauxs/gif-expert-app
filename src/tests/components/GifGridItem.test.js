import React from 'react'
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe('Pruebas en <GifGridItem/>', () => {


    const title = "El test"
    const url = "https:/laimagentest.com"
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('should: mostrar correctamente el componente GifGridItem ', () => {

        expect(wrapper).toMatchSnapshot()
    })


    test('should: mostrar el parrafo', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    })

    test('should: mostrar la imgagen', () => {

        const img = wrapper.find('img')
        // console.log(img.html())
        // console.log(img.props())

        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })


    test('should: tener la clase animate__bounce', () => {
        const div = wrapper.find('div')
        const className = div.prop('className')
        
        expect(className.includes("animate__bounce")).toBe(true)
    })

})
