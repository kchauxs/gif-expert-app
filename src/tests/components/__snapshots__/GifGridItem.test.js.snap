// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> should: mostrar correctamente el componente GifGridItem  1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="El test"
    src="https:/laimagentest.com"
  />
  <p>
    El test
  </p>
</div>
`;
